### PR TITLE
docs: migration guide to v3

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

**Comment by csviri** [comment]
This is quick list, will add continuously other other if I find one.

File: docs/documentation/v3-migration.md [hunk 1]
```diff
@@ -6,3 +6,34 @@ permalink: /docs/v3-migration
 ---
 
 # Migrating from v2 to v3
+
+Version 3 introduces some breaking changes to the main and secondary APIs, however the migration to these changes are
```
Comment (metacosm) [inline docs/documentation/v3-migration.md:10]: What's the `secondary` API?

File: docs/documentation/v3-migration.md [hunk 2]
```diff
@@ -6,3 +6,34 @@ permalink: /docs/v3-migration
...
+trivial.
+
+## Reconciler
+
+- [`Reonciler`](https://github.com/java-operator-sdk/java-operator-sdk/blob/67d8e25c26eb92392c6d2a9eb39ea6dddbbfafcc/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java#L16-L16)
```
Comment (metacosm) [inline docs/documentation/v3-migration.md:15]:
```suggestion
- [`Reconciler`](https://github.com/java-operator-sdk/java-operator-sdk/blob/67d8e25c26eb92392c6d2a9eb39ea6dddbbfafcc/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java#L16-L16)
```

File: docs/documentation/v3-migration.md [hunk 3]
```diff
@@ -6,3 +6,34 @@ permalink: /docs/v3-migration
...
+  can throw checked exception (not just runtime exception), and that also can be handled by `ErrorStatusHandler`.
+- `cleanup` method is extracted from the `Reconcile` to a
```
Comment (metacosm) [inline docs/documentation/v3-migration.md:17]:
```suggestion
- `cleanup` method is extracted from the `Reconciler` interface to a
```

File: docs/documentation/v3-migration.md [hunk 4]
```diff
@@ -6,3 +6,34 @@ permalink: /docs/v3-migration
...
+  separate [`Cleaner`](https://github.com/java-operator-sdk/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Cleaner.java)
+  interface. It was articulated also until now, that finalizers only makes sense that the `Cleanup` is implemented, from
```
Comment (metacosm) [inline docs/documentation/v3-migration.md:19]: I don't understand what this: "It was articulated also until now…" means.

File: docs/documentation/v3-migration.md [hunk 5]
```diff
@@ -6,3 +6,34 @@ permalink: /docs/v3-migration
...
+  now finalizer is only added if the `Reconciler` implements this interface (or has managed dependent resources
+  implementing `Deleter` interface, see dependent resource docs).
+- [`Context`](https://github.com/java-operator-sdk/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Context.java#L9-L9)
+  object of `Reconciler` now takes the Primary resource as parametrized type: `Context<MyCustomResource>`.
+- [`ErrorStatusHandler`](https://github.com/java-operator-sdk/java-operator-sdk/blob/67d8e25c26eb92392c6d2a9eb39ea6dddbbfafcc/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/ErrorStatusHandler.java)
+  result changed, it's functionally has been extended, now can prevent Exception to be retries and handle checked
```
Comment (metacosm) [inline docs/documentation/v3-migration.md:25]:
```suggestion
  result changed, it functionally has been extended to now prevent Exception to be retried and handles checked
```

File: docs/documentation/v3-migration.md [hunk 6]
```diff
@@ -6,3 +6,34 @@ permalink: /docs/v3-migration
...
+
+
+## Event Sources
+
+- Event Sources are now registered with a name. But [utility method](https://github.com/java-operator-sdk/java-operator-sdk/blob/92bfafd8831e5fb9928663133f037f1bf4783e3e/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/EventSourceInitializer.java#L33-L33) 
+  is available to make it easy to [migrate](https://github.com/java-operator-sdk/java-operator-sdk/blob/92bfafd8831e5fb9928663133f037f1bf4783e3e/sample-operators/webpage/src/main/java/io/javaoperatorsdk/operator/sample/WebPageStandaloneDependentsReconciler.java#L51-L52)
+  to a default name.  
+- [InformerEventSource](https://github.com/java-operator-sdk/java-operator-sdk/blob/92bfafd8831e5fb9928663133f037f1bf4783e3e/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/InformerEventSource.java#L75-L75)
+  constructor changed to reflect additional functionality in a non backwards compatible way. But should be trivial to 
+  migrate based on the new API.
```
Comment (metacosm) [inline docs/documentation/v3-migration.md:36]: Saying it's trivial doesn't really help users. We should probably show how to migrate.

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/ControllerConfiguration.java [hunk 7]
```diff
@@ -16,8 +16,8 @@
 
   /**
    * Optional finalizer name, if it is not provided, one will be automatically generated. Note that
-   * finalizers are only added when Reconciler implement {@link Cleaner} interface, or at least one
-   * managed dependent resource implement
+   * finalizers are only added when Reconciler implement {@link Cleaner} interface and/or at least
+   * one managed dependent resource implement
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/ControllerConfiguration.java:20]:
```suggestion
   * one managed dependent resource implements the
```

**Review by metacosm — CHANGES_REQUESTED** [review]

File: docs/documentation/v3-migration.md (hunk 4 above)
Comment (csviri) [reply docs/documentation/v3-migration.md]: changed.

File: docs/documentation/v3-migration.md (hunk 6 above)
Comment (csviri) [reply docs/documentation/v3-migration.md]: updated. added link for usage, and the configuration. The API is self explanatory, and should be very easy to migrate based on those.

**Review by metacosm — APPROVED** [review]
Thank you!

**Pull request merged**